Dockerfile

```diff
@@ -36,7 +36,9 @@ RUN curl -fsSL https://mafft.cbrc.jp/alignment/software/mafft-7.475-linux.tgz \
 WORKDIR /build/RAxML
 RUN curl -fsSL https://api.github.com/repos/stamatak/standard-RAxML/tarball/v8.2.12 \
   | tar xzvpf - --no-same-owner --strip-components=1
-RUN make -f Makefile.AVX.PTHREADS.gcc   # AVX should be widely-supported enough
+# AVX should be widely-supported enough
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then make -f Makefile.AVX.PTHREADS.gcc; fi
```

Comment:
I noticed that [raxml_8.2.12+dfsg-4_**arm64**.deb](https://ubuntu.pkgs.org/20.04/ubuntu-universe-arm64/raxml_8.2.12+dfsg-4_arm64.deb.html) is available, which means it can be done.

Digging deeper on that page, the **Package Homepage** turns up a [link to a git repo](https://code.launchpad.net/~git-ubuntu-import/ubuntu/+source/raxml/+git/raxml):

```
git clone https://git.launchpad.net/~git-ubuntu-import/ubuntu/+source/raxml
```

The contents are similar to https://github.com/stamatak/standard-RAxML but importantly there is an additional [`debian/`](https://git.launchpad.net/~git-ubuntu-import/ubuntu/+source/raxml/tree/debian) directory.

Some lines in the file [`debian/rules`](https://git.launchpad.net/~git-ubuntu-import/ubuntu/+source/raxml/tree/debian/rules) are useful:

```make
ifeq (amd64,$(DEB_TARGET_ARCH_CPU))
##MFILES:=Makefile.gcc \ # we use a wrapper to the faster versions instead of shipping the "plain" version
MFILES =Makefile.PTHREADS.gcc \
	Makefile.SSE3.PTHREADS.gcc \
	Makefile.AVX.PTHREADS.gcc
MFLAGS=
else
MFILES = Makefile.AVX.PTHREADS.gcc
MFLAGS = CFLAGS="$(shell dpkg-buildflags --get CFLAGS) -DSIMDE_ENABLE_OPENMP -fopenmp-simd -O3"
endif
```

This means `Makefile.AVX.PTHREADS.gcc` is used to run on `arm64`, except with custom `CFLAGS` that override the [`CFLAGS` in the Makefile](https://github.com/stamatak/standard-RAxML/blob/a33ff40640b4a76abd5ea3a9e2f57b7dd8d854f6/Makefile.AVX.PTHREADS.gcc#L6) where [`-msse3` was causing our `arm64` compilation error](https://github.com/nextstrain/docker-base/pull/48#issuecomment-1154560485).

The file [`debian/patches/hardening`](https://git.launchpad.net/~git-ubuntu-import/ubuntu/+source/raxml/tree/debian/patches/hardening) is even more telling:

```diff
-CFLAGS := -D_USE_PTHREADS  -D__SIM_SSE3 -O2 -D_GNU_SOURCE -msse3 -fomit-frame-pointer -funroll-loops  -D__AVX #-Wall -pedantic -Wunused-parameter -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wunused-value -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport  -Wunused  -Wunused-function  -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes  -Wdeclaration-after-statement -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef -Wbad-function-cast
+CFLAGS += -D_USE_PTHREADS  -D__SIM_SSE3 -O2 -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops  -D__AVX #-Wall -pedantic -Wunused-parameter -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wunused-value -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport  -Wunused  -Wunused-function  -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes  -Wdeclaration-after-statement -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef -Wbad-function-cast
 
+ifeq (amd64,$(shell dpkg-architecture -q DEB_TARGET_ARCH_CPU))
+	CFLAGS += -mavx -msse3
+endif
```

This patch updated the runner to only use `-msse3` on `amd64`, but the change isn't reflected in the Makefile that we download from GitHub which is why we still see the `-msse3` error.

I suspect that this should work for us using the same Makefile: `CFLAGS="-D_USE_PTHREADS -D__SIM_SSE3 -O2 -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -D__AVX"`

Reply:
It worked in a test run on 6ffbdbad142cfdb597466408e76a0769618f5ced (`devel/build` succeeded on the push-triggered run).

Added to this PR in d2cf41abcaa149cfd13871c67c5d773afd6892cc.

Reply:
Compilation with those flags fail. I tried using both the existing reference to https://api.github.com/repos/stamatak/standard-RAxML/tarball/v8.2.12 and the Ubuntu git repo at https://git.launchpad.net/~git-ubuntu-import/ubuntu/+source/raxml. [Both fail with](https://github.com/nextstrain/docker-base/runs/8147469773?check_suite_focus=true#step:8:4116):

```
[linux/arm64 builder 6/6] RUN if [ "linux/arm64" = "linux/arm64" ]; then make -f Makefile.AVX.PTHREADS.gcc CFLAGS="-D_USE_PTHREADS  -D__SIM_SSE3 -O2 -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops  -D__AVX"; fi
rm -f *.o raxmlHPC-PTHREADS-AVX
gcc  -D_USE_PTHREADS  -D__SIM_SSE3 -O2 -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops  -D__AVX   -c -o axml.o axml.c
axml.c:70:10: fatal error: xmmintrin.h: No such file or directory
 #include <xmmintrin.h>
          ^~~~~~~~~~~~~
compilation terminated.
make: *** [<builtin>: axml.o] Error 1
```

Reply:
Ok, using the flags alone wasn't enough. There's actually a GitHub PR created by the patch author for similar intents, and it works after some additional changes: https://github.com/stamatak/standard-RAxML/pull/50#issuecomment-1235769882

Reply:
[`linux/arm64` RAxML build is now successful](https://github.com/nextstrain/docker-base/runs/8161756817?check_suite_focus=true#step:8:3818).